src/ImageGallery.jsx

```diff
@@ -258,14 +258,16 @@ export default class ImageGallery extends React.Component {
       slideDuration,
       startIndex,
       thumbnailPosition,
+      showThumbnails,
     } = this.props;
     const { currentIndex } = this.state;
     const itemsSizeChanged = prevProps.items.length !== items.length;
     const itemsChanged = !isEqual(prevProps.items, items);
     const startIndexUpdated = prevProps.startIndex !== startIndex;
     const thumbnailsPositionChanged = prevProps.thumbnailPosition !== thumbnailPosition;
+    const showThumbnailsChanged = prevProps.showThumbnails !== showThumbnails;
 
-    if (thumbnailsPositionChanged) {
+    if (thumbnailsPositionChanged || showThumbnailsChanged) {
```

Comment:
This should actually call `handleResize` not re-initialize the `resizeObserver`. the observer is only reinitialized for the slides when they unmount and remount. I need to document that better.

you can place this logic a few lines under in https://github.com/xiaolin/react-image-gallery/blob/4f295c4248ac2ffac0895bba8e8d251c28283b0c/src/ImageGallery.jsx#L277

Reply:
sure, updating `showThumbnails` is triggering `handleResize` instead of reinit observer now

Reply:
@xiaolin could you please update the version in npm?

Reply:
I'll release a new version tonight.

Reply:
Released in version `1.04`